scripts/data_prep/convert_text_to_mds.py

```diff
@@ -0,0 +1,329 @@
+        names = [
+            name for name in object_store.list_objects(folder_prefix)
+            if name.endswith('.txt')
+        ]
+    else:
+        # input_folder is a local folder
+        names = [
+            text_file for dirpath, _, _ in os.walk(input_folder)
+            for text_file in glob(os.path.join(dirpath, '*.txt'))
+        ]
+    # return names, sizes
+    print(f'Found {len(names)} text files at {input_folder}')
+
+    return names
+
+
+def get_task_args(
+    object_names: List[str],
+    output_root: str,
+    input_folder: str,
+    n_groups: int,
+    tokenizer_name: str,
+    concat_tokens: int,
+    eos_text: str,
+    bos_text: str,
+    no_wrap: bool,
+    compression: str,
+    max_workers: int,
+) -> Iterable:
+    objs_per_group = math.ceil(len(object_names) / n_groups)
+    for group, i in enumerate(range(0, len(object_names), objs_per_group)):
+        output_subdir = os.path.join(output_root, str(group))
+        yield (
+            object_names[i:i + objs_per_group],
+            output_subdir,
+            input_folder,
+            tokenizer_name,
+            concat_tokens,
+            eos_text,
+            bos_text,
+            no_wrap,
+            compression,
+            max_workers,
+        )
+
+
+def download_and_convert(
+    file_names: List[str],
+    output_folder: str,
+    input_folder: str,
+    tokenizer_name: str,
+    concat_tokens: int,
+    eos_text: str,
+    bos_text: str,
+    no_wrap: bool,
+    compression: str,
+    max_workers: int,
+):
+    object_store = maybe_create_object_store_from_uri(input_folder)
+    _, _, folder_prefix = parse_uri(input_folder)
+
+    # Download file_names
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        downloading_iter = DownloadingIterable(file_names, folder_prefix,
+                                               tmp_dir, object_store)
+        tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+        tokenizer.model_max_length = 5000000000
+
+        # Use the ConcatTokensDataset from LLM-foundry to concatenate sequences of tokens up to the maximum sequence length
+        dataset = ConcatTokensDataset(
+            hf_dataset=downloading_iter,
+            max_length=concat_tokens,
+            tokenizer=tokenizer,
+            eos_text=eos_text,
+            bos_text=bos_text,
+            no_wrap=no_wrap,
+        )
+
+        loader = build_dataloader(dataset=dataset, batch_size=512)
+        samples = generate_samples(loader)
+        columns = {'tokens': 'bytes'}
+
+        print(f'Converting to MDS format...')
+        with MDSWriter(out=output_folder,
+                       columns=columns,
+                       max_workers=max_workers,
+                       compression=compression) as out:
+            for sample in tqdm(samples):
+                out.write(sample)
+
+
+def is_remote_path(path: str) -> bool:
+    backend, bucket, _ = parse_uri(path)
+    return backend != '' or bucket != ''
```

Comment:
I arrived to that logic since a local path (according to the doc string of parse_uri) is where `backend == '' and bucket == ''`, so `is_remote_path => not (backend == '' and bucket == '') => backend != '' or bucket != ''`

Reply:
huh, fair enough. not sure if the upstream implementation is wrong or not...but we def need a backend and a bucket for our purposes here, so lets just check that.